woocommerce/google-listings-and-ads
# Account Review Status

### Changes proposed in this Pull Request:

Closes part of #1165.

This PR implements the new Design requirements as well as some refactor in the Status Box and the Request Review.

[Figma Design](https://www.figma.com/file/eQ9O4m2flzrcAiDMXkOW0m/Google-Listings-%26-Ads-v1.x?node-id=6956%3A119421)
[Design Thread](https://wp.me/pcTzPl-ps)
[Project Thread](https://googleforwooextension.wordpress.com/2021/12/06/project-thread-gmc-account-review-request/)

### Detailed test instructions:

1. Checkout PR
2. Go to `src/API/Site/Controllers/MerchantCenter/AccountReviewController.php::get_review_read_callback()`  method in your editor.
3. In line 60 you will need to change 'status' => 'DISAPPROVED', to the different statues to test the behaviour. 
4. Go to Product Feed - `/wp-admin/admin.php?page=wc-admin&issueType=account&path=%2Fgoogle%2Fproduct-feed` to see the result
5. Check as well that Sync and feed still looks like before. Green icon and green text. 
6. Notice as well that the icons could vary from the design, but the meaning and color of the icon should be the same.

**DISAPPROVED** 

<img width="889" alt="Screenshot 2022-04-14 at 12 34 03" src="https://user-images.githubusercontent.com/5908855/163346782-204eb25a-3f04-48f9-897d-c9a6dd0eb61c.png">

Status box should look like this with Status Disapproved in red and the icon in red as well. Also you can see the Notice in Product Issues tab with the review button.

**UNDER_REVIEW** 

<img width="873" alt="Screenshot 2022-04-14 at 12 38 57" src="https://user-images.githubusercontent.com/5908855/163347595-75011e78-893b-4052-918c-8c212c1e4ef2.png">

Status box should look like this with Status Under Review in black and the icon in yellow. Also you can see the Notice in Product Issues tab WITHOUT the review button.

**PENDING_REVIEW** 

<img width="1133" alt="Screenshot 2022-04-14 at 12 39 24" src="https://user-images.githubusercontent.com/5908855/163347669-01ada091-8026-469e-9c8f-4d4cba78658d.png">

Status box should look like this with Status Pending Review in black and the icon in yellow. Also you can see the Notice in Product Issues tab WITHOUT the review button.

**APPROVED** 

<img width="1009" alt="Screenshot 2022-04-14 at 12 40 08" src="https://user-images.githubusercontent.com/5908855/163347782-82dd3253-7e6a-4e8d-8076-a4ad885682c0.png">

Status box should look like this with Status Approved in green and the icon in green as well. Also you CANNOT see the Notice in Product Issues tab.

**WARNING** 

<img width="1004" alt="Screenshot 2022-04-14 at 12 39 48" src="https://user-images.githubusercontent.com/5908855/163347736-834d1aa5-a437-42c7-a86f-fdac71f95964.png">


Status box should look like this with Status Warning in black and the icon in yellow. Also you can see the Notice in Product Issues tab WITH the review button.

* In this last Status go to the dev tools editor. Inspecting the description (the text that is in light grey) then add random text forcing it to have multiple lines. Check that the multiline version looks accurate with the design.

<img width="1023" alt="Screenshot 2022-04-14 at 12 40 43" src="https://user-images.githubusercontent.com/5908855/163348760-c365642d-44ab-4b5b-8195-1f1b7a0d4433.png">


### Additional details:

- Created `<Status />` component avoiding duplication. 
- Refactor `<StatusBox />`  Feed moving the render and the function into their own file.
- Refactor `<StatusBox />`  Sync moving the render and the function into their own file.
- Created `<AccountStatus />` rendering the Account Status info.
- Styling
- Tests
- Adjust Request Review tests and some minor logic. Especially the BLOCKED status that will be deprecated because now Google includes the Cooldown period in  the response so we can handle this in a better way now. (To be implemented in Future PR)
